phoenix-core/src/main/java/org/apache/phoenix/compile/FromCompiler.java

```diff
@@ -1171,14 +1183,69 @@ public ColumnRef resolveColumn(String schemaName, String tableName, String colNa
                 TableRef tableRef = isLocalIndex ? super.getTables().get(0) : super.resolveTable(schemaName, tableName);
                 if (tableRef.getTable().getIndexType() == IndexType.LOCAL) {
                     try {
-                        TableRef parentTableRef = super.resolveTable(
-                                tableRef.getTable().getSchemaName().getString(),
-                                tableRef.getTable().getParentTableName().getString());
+                        TableRef parentTableRef = null;
+                        if (tableName == null) {
+                          // No table specified
+                          Iterator<TableRef> iterator = localIndexDataTables.iterator();
+                          while (iterator.hasNext()) {
+                              TableRef searchTableRef = iterator.next();
```

Comment:
Could be clubbed to `for (TableRef localIndexDataTableRef : localIndexDataTables)`